lib/vector-buffers/src/variants/disk_v2/tests/basic.rs

```diff
@@ -67,6 +67,17 @@ async fn basic_read_write_loop() {
     .await;
 }
 
+// run `cargo test --no-run -p vector-buffers`
+// copy output path should look like (target/debug/deps/vector_buffers-xxxx)
+// to test run:
+// ```
+// for i in $(seq 1 10000); do
+//   echo $i
+//   ./target/debug/deps/vector_buffers-xxxx reader_exits_cleanly_when_writer_done_and_in_flight_acks || break
+// done
+// ```
+// This test fails once in ~3000 runs
```

Comment:
This too complex. Instead: 

```suggestion
cargo nextest run -p vector-buffers \
  --filter reader_exits_cleanly_when_writer_done_and_in_flight_acks \
  --retries 10000
```

BTW it's worth checking the above to ensure this is flakey when run via `nextest` as well.